numpy/_core/src/common/npy_import.h

```diff
@@ -16,17 +89,28 @@
  * @param attr module attribute to cache.
  * @param cache Storage location for imported function.
  */
-static inline void
-npy_cache_import(const char *module, const char *attr, PyObject **cache)
-{
-    if (NPY_UNLIKELY(*cache == NULL)) {
-        PyObject *mod = PyImport_ImportModule(module);
-
-        if (mod != NULL) {
-            *cache = PyObject_GetAttrString(mod, attr);
-            Py_DECREF(mod);
+static inline int
+npy_cache_import_runtime(const char *module, const char *attr, npy_runtime_import *cache) {
+    if (cache->initialized) {
```

Comment:
Before diving deeper, can you explain why we add the `->initialized` here.  I.e. is it not possible to do:
```
if (*obj != NULL) {
    return NULL;
}
else {
     if (!npy_atomic_load_pyobj(obj)) {
...
```
It seems easier to just call `npy_cache_import("module", "attr", & npy_runtime_imports.array_function_formatter)` and then not having to worry about that `.obj` later on.
(Also any late-initialized object could be inside the same struct, although I am not sure we have many.)

Not a huge thing, but it's nice not to have to worry about the `.obj` int he code using these.

Reply:
Yeah, I think I can refactor things to avoid the struct and use pointer loads. I'll go ahead and do that along with Sam's corrections.